src/Annotation/AttributesHydratorTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Annotation;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\Common\Util\Inflector;
+
+/**
+ * Hydrates attributes from annotation's parameters.
+ *
+ * @internal
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+trait AttributesHydratorTrait
+{
+    /**
+     * @var array
+     */
+    public $attributes = [];
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $values = [])
```

Comment:
Because it's one more function call and more one level of indirection for no benefit. It's unusual to add a constructor in a trait, but for an annotation like this it's legit. And it's, why this trait is marked `@internal`.

Reply:
@Simperfit do you prefer the strategy in the last commit? It is more explicit, SOLID and don't add an extra call.